### PR TITLE
Fix kqueue for platform where C's long is i32

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
This for the master branch.

Updates #946.

### CODE REVIEW

File: src/sys/unix/kqueue.rs [hunk 1]
```diff
@@ -74,7 +74,9 @@ impl Selector {
     ) -> io::Result<bool> {
         let timeout = timeout.map(|to| libc::timespec {
             tv_sec: cmp::min(to.as_secs(), time_t::max_value() as u64) as time_t,
-            tv_nsec: libc::c_long::from(to.subsec_nanos()),
+            // On some platforms C's long is i32, for those we need to ensure
+            // `subsec_nanos` doesn't overflow.
+            tv_nsec: cmp::min(to.subsec_nanos(), libc::c_long::max_value() as u32) as libc::c_long,
```
Comment (asomers) [inline src/sys/unix/kqueue.rs]: `libc::c_long::max_value() as u32` will overflow on many platforms.  How about `i32::max_value() as u32` instead?

Comment (Thomasdezeeuw) [reply src/sys/unix/kqueue.rs]: I was wondering this myself as well (https://github.com/tokio-rs/mio/pull/948#discussion_r285062700), I think `i32::max_values` would do the trick.

**Comment by Thomasdezeeuw** [comment]
@asomers Fixed the overflow.

**Comment by Thomasdezeeuw** [comment]
@asomers I forgot it can only be less than one billion, I've changed the code to your suggestion.

File: src/sys/unix/kqueue.rs [hunk 2]
```diff
@@ -74,7 +74,11 @@ impl Selector {
     ) -> io::Result<bool> {
         let timeout = timeout.map(|to| libc::timespec {
             tv_sec: cmp::min(to.as_secs(), time_t::max_value() as u64) as time_t,
-            tv_nsec: libc::c_long::from(to.subsec_nanos()),
+            // `Duration::subsec_nanos` is guaranteed to be less than one
+            // billion (the number of nanoseconds in a second), making the
+            // cast to i32 safe. The cast itself is needed for platforms
+            // where C's long is only 32 bits.
```
Comment (asomers) [inline src/sys/unix/kqueue.rs]: The comment isn't quite right.  The cast is only needed for platforms where C's long is `i32`.  No cast is needed where long is `u32`.

Comment (Thomasdezeeuw) [reply src/sys/unix/kqueue.rs]: Isn't C `long` always signed? According to https://en.wikipedia.org/wiki/C_data_types: "Long signed integer type. Capable of containing at least the [−2,147,483,647, +2,147,483,647] range;[3][4] thus, it is at least 32 bits in size."

Comment (asomers) [reply src/sys/unix/kqueue.rs]: Yes it is always signed, except when I'm being stupid.

**Pull request merged**